source/docs/software/wpilib-tools/robot-simulation/introduction.rst

```diff
@@ -0,0 +1,45 @@
+Introduction to Robot Simulation
+================================
+
+Often a team may want to test their code without having an actual robot available. WPILib provides teams with the ability to simulate the robot's Hardware Abstraction Layer (HAL) using simple gradle commands.
+
+Enabling Desktop Support
+------------------------
+
+Before you can start robot simulation, it's required to have desktop support enabled. Desktop support can be enabled via checking the  "Enable Desktop Support Checbox" when creating your robot project, or via running "WPILib: Change Desktop Support Enabled Setting" from the command palette. If none of the already listed options work, please `report a bug <https://github.com/wpilibsuite/allwpilib/issues>`__.
+
+.. image:: images/vscode-desktop-support.png
+   :alt: Enabling desktop support through VSCode
+
+
+.. image:: images/vscode-desktop-support-manual.png
+   :alt: Manually enabling desktop support through VSCode command-palette
+
+Desktop support can also be enabled by manually editing your ``build.gradle`` file located at the root of your robot project. Simply change ``includeDesktopSupport = false`` to ``includeDesktopSupport = true``
+
+.. code-block:: text
+
+   def includeDesktopSupport = true
+
+.. important:: It is important to note that enabling desktop/simulation support can have unintended consequences. Not all vendors will support this option, and code that uses their libraries may even crash when attempting to run simulation!
```

Comment:
We should also mention that for C++ a native compiler is required (for Windows, VS2019 must be installed).

Reply:
Oof. That's a hefty dependency.